codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -138,9 +152,20 @@ private static ByteBuf encodeConnectMessage(
             payloadBufferSize += 2 + passwordBytes.length;
         }
 
-        // Fixed header
+        // Fixed and variable header
         byte[] protocolNameBytes = mqttVersion.protocolNameBytes();
-        int variableHeaderBufferSize = 2 + protocolNameBytes.length + 4;
+        ByteBuf propertiesBuf = encodePropertiesIfNeeded(
+                mqttVersion,
+                byteBufAllocator,
+                message.variableHeader().properties());
+        int variableHeaderBufferSize = 2 + protocolNameBytes.length + 4 + propertiesBuf.readableBytes();
+        final ByteBuf willPropertiesBuf;
```

Comment:
you will need to call `willPropertiesBuf.release()` later as you not transfer ownership